### ISSUE TITLE
[SUGGESTION] The Wizard's Badmin Gauntlet (Infinity Gauntlet) + Declaration of Wizardry Warfare!

### ISSUE THREAD

**Issue by Manly-Dwarf** [HippieStation/HippieStationdeprecated2020#9145]
## Badmin Gauntlet
(it's actually just the Infinity Gauntlet but with some things changed so we don't get sued by disney and we're within grounds of parody)
Also I'll be the one coding this, mainly. At least until I realize I have no idea what I'm doing and ask for help.
![thanos](https://user-images.githubusercontent.com/33744849/45607414-5ef79500-ba1a-11e8-9283-cc678bdab25d.gif)

[Implementation]: 
The Badmin Gauntlet will be an "Artifact of Power" sold to the Wizard, similar to the Book of Darkness. It should cost 10 points, i.e all of the Wizards points, and simply spawns a paper. The paper will explain that the Badmin Gauntlet is a _declaration of war_ and spawn a special box. If they hit the Spell Book with the box, they will get all of their points back and they can go back to normal. If they open the special box, an announcement (similar to the Nanosuit) will be made along with a message for a declaration of war from the Wizard's federation. The wizard will be equipped with the Badmin Gauntlet and some base spells that are more meant for general versatility and power _before_ he gets the stones. When he gets the stones, some things change.
The basis of the idea is that the Wizard is able to cast spells and dish out huge damage with the gauntlet simultaneously, creating an incredibly overpowered antagonist truly deserving of a Declaration of War.

### The Gauntlet
The Gauntlet will take up both a glove spot and your right hand, where both are nodrop. The gloves will protect you from shocks and fire. The Gauntlet in your right hand will function as your means of actually dealing damage, swapping modes with the Gauntlet, and so on.
Intents in every mode generally do something different. But, Gauntlet starts off with no extra "modes", it's just your Gauntlet. "Modes" are unlocked by grabbing the extra stones which makes you gradually more powerful.

All of the modes are completely possible with the current code and actually would not require too much fiddling. It's not my fault it'll end up being around a thousand lines long regardless, Carbon.

Mode 1: Default (No Stones) / Syndie Stone
The Default Gauntlet is available only when you don't have the Antag Stone. It boasts Martial Arts combos, huge damage, 50% ranged deflects, and the ability to punch down walls with some startup time. Using help intent on someone who is downed will have you try to break one of their legs, but if they're standing you'll merely shove them.
Once you get the Syndie Stone, Default is made much stronger. Punching down walls is now instant, combos deal much more damage, and you will now stun on punch similar to Hulks. On top of that, using help on someone who is downed will let you rip off their leg and use it as a long ranged weapon, or even throw it to deal more damage.

Mode 2: Bluespace Stone
The Bluespace Stone is mostly helpful for getting around quickly. Most of its use comes from the spells it grants, but you can still use the gauntlet for combat purposes.
Help Intent: Warp Whistle effect. Can be used on yourself, or your enemies!
Harm Intent: Quickly warp where you were clicking. Drains some stamina.
Grab Intent: No changes. Suggest some!
Disarm intent: Knocks their weapons straight into your backpack! If you don't have the room, teleports said item to a random room.

Mode 3: Server(? name W.I.P.) Stone.
Not much, but worthwhile.
Using the mode on walls will turn the walls into completely random objects. If used on things like Cloth, it can turn it into golems. Useful for amassing armies out of the salted ghosts.
All intents but Harm: Fire a bolt of animation.
Harm Intent: Staff of Chaos. As things go with the Staff of Chaos, this will either end amazingly or terribly.

Mode 4: Clown Stone.
Because who has more soul on the station than the Clown? Mostly good for traps.
Help Intent: Pie Cannon. The pies contain random chemicals in them, so it's ideal that you don't fire this in your face. Very small chance to blast off a clown car.
Harm Intent: Spawns The Traps!
Grab Intent: Infects the target's soul with Clownington's Degenerative Disorder, causing them to slowly become a clown who can barely fight back. There is no cure, unlike the Server Stone's effects.
Disarm Intent: Sprays Water (not lube, that'd just get annoying) all over the floor.

Mode 5: Lag Stone.
The bane of Carbonhell's existence, the Lag Stone mode mostly exists to help you get out of bad situations fast. It simply speeds you up while in use, slowly draining your stamina similar to Changeling's Strained Muscles. But you can still pack a punch.
Help Intent: Attempts to teleport you back to a set point on the station, and heals you drastically. You will be stunned for a great time after using it, and it has some startup, so it is NOT for free. Alternatively, you can use this alongside GBJ to create a safehouse, or a massive prison if used on your foes! The main use for the mode, outside of running away. You may say this is more fitting for the Bluespace Stone butidontcarefuckyou
Harm Intent: Space Ninja's weird teleport slicey thing. I don't know the name or the rules.
Grab Intent: Fires a projectile, and swaps places with the victim!
Disarm Intent: Shoots a projectile that will create a ton of dizzying effects on the target.

"Mode" 6: Powergame Stone.
Name suggested by Maroon, who is a moron. The joke being that Powergame is all in the mind.
All Intents: Lasers, because if there's anything that reminds you of powergame it's tase + lase.

### Spells
Obviously, Wiz starts off with the classic MM and EJ. He also starts off with three other spells:
Shockwave, Super Punch, Regeneration and Charge. What do they have to do with Thanos? Beats me, but he has super strength and borderline immortality so give me some creative liberty.
Each and every spell (for now) will be listed below:
**Shockwave:** Spawns a massive shockwave, knocking opponents away from the user and to the ground for a short time. A useful "get off of me" tool.
Is buffed up immensely once you have the Syndie Stone, has a possibility of setting people on fire.
**Super Punch:** Punch so hard you send a massive wave flying forward incredibly quickly, dealing heavy damage on contact.
Once you have the Syndie Stone, you will [learn how to use your biceps](https://github.com/HippieStation/HippieStation/pull/5091), which makes the already powerful shot even better, especially because your biceps are much better than those puny normal Wizards who can only launch people like three tiles away.
**Charge:** No, not like the recharge spell, this is just a rod-form-esque bull rush forward. Anyone who gets hit gets pummeled.
Once you have the Syndie Stone, it'll let you smash through walls and go even faster while pummeling harder. Once in use, you'll also be invincible to all damage in its duration.
**Regeneration:** If you had to pussy out of a fight, then this is a good way to recuperate. Requires that you stand still for an extended time, but will heal very admirably without the need of other chemicals or those pesky doctors.

### Spells - Syndie Stone
See above.

### Spells - Bluespace Stone
**Bluespace Jaunt:** Replaces Ethereal Jaunt. It moves much quicker, allowing you to cover way more space than usual, and still works when stunned.
**Bluespace Blink:** A blink spell with much more range, but leaves behind a tracer of where you went. Don't worry, you can breathe in space.
**Portal Shield:** Summons a shield in your left hand which will negate any and all damage coming your way for 15 seconds, or until you get hit 5 times. Attacks are sent wildly across bluespace.
### Spells - Server Stone
**Gondola Morph:** Upon contact with a target, turns them into a gondola! If the gondola is killed, they will pop back out and return to their regular selves.
**Reality Distortion:**
https://github.com/HippieStation/HippieStation/blob/c87165263ea55d1a0b1a82f52adbcd1f67729a2c/code/modules/spells/spell_types/spacetime_distortion.dm
**Summon GBJ:** Creates an inflatable jail exactly where you stand. This can either be used as a safe home for recovery, or as a prison for victims. Once this spell is used, it's gone forever.
![inflatable house](https://user-images.githubusercontent.com/33744849/45614408-7133fc00-ba37-11e8-8075-251a7084a414.jpg)

### Spells - Honk Stone
**Rise Up:** Returns a dead body to a completely subservient cluwne! Obviously worse than a Golem, but cheaper.
**Prankster's Delusion:** Causes those around you to see others as a clumsy clown, including yourself! Now how will they know who is who?

### The Stones
Once War is openly declared, the stones will be delivered to certain crewmembers or scattered about. Crewmembers in possession of a stone will radiate a glow or aura that lets everyone know of their importance, including the Wizard if he swaps modes on his Gauntlet.
Stones _should_ attempt to roll for the person with the highest chance of getting said stone first, i.e the Clown will be rolled for the Honk Stone first before the Captain. Everyone starts off with a flat 10% chance for each stone, but can only get one stone. On top of that, every stone also has a modifier giving some jobs a better shot at it. If every single roll fails, the Stone should randomly respawn in maint similar to a Nuke Disk.
If someone bearing a stone is killed, it is immediately removed from their possession and dropped on the ground ripe for the taking. Cremating or spacing a stone will merely have it respawn on the station, again like the Nuke Disk.
Modifiers are as follows:
**Syndie Stone:**
HoS: 70% chance. Captain: 45% chance. Security Officers: 20% chance. Head of Personnel: 15% chance.
The person who has the Syndie Stone gains a martial art similar to the Default MA, and passively recovers from injuries over time.

**Bluespace Stone:**
RD: 60% chance. Scientists: 20% chance. Mime: 15% chance (because he's all about running away!)
The person who has the Bluespace Stone gains a considerably weaker Bluespace Blink, and gains the Bluespace Prophet brain "trauma."

**Server Stone:**
CE: 60% chance. Librarian: 45% chance. Engineers: 30% chance.
The person who has the server stone basically just has telekinesis and x-ray vision because I don't have any ideas and am a boring person.

**Honk Stone:**
Clown: 90% chance. Librarian: 20% chance. Mime: 0% chance.
The person who has the Honk Stone can summon banana peels out of thin air, prime for the chucking.

**Lag Stone:**
Quartermaster: 40% chance. Other Cargonians (not miners): 20% chance.
The bearer of the Lag Stone gains the same ability as the Gauntlet's Help Intent, but it only works on themselves and whoever they have grabbed. It also has longer startup.

**Powergame Stone:**
Captain: 70% chance. Janitor: 25% chance.
The bearer of the Powergame Stone gains the Recharge spell, as well as being able to communicate with ghosts via Scrying Orb-like spells.

Once all stones are handed out, the crew is given around 5-10 minutes to prepare for a Wizard attack. Cargo is also rewarded with additional points. **Because war is declared, I strongly suggest that if this ever gets merged, admins ban any sort of grief. No reason to act like a dick in decwar.**

## Once all stones have been collected by the Wiz they're meant to be practically unstoppable. And of course, they get the famed snap to kill half of the station population.

[Why]: # How does your feature improve the game anyhow? Why should it be added?
Because Wizard has no real means of declaration of war. Simply put, he only has Summon Guns and Magic, which aren't actually challenges for the Wizard. They merely make the crew murderbone each other and drive admins to drink. Any replacement for that is ideal, especially after /tg/ nerfed it into the ground.
Really, this whole idea is just a fun/funny way of seeing what could be done with Declaration of War with a Wizard. DecWar is underappreciated for how much fun or chaotic it can be, thanks to our population having been in decline for a while and Nuke Ops barely happening any more. But now that we have a population, I think it's time we could add something to Wizard for that purpose.
Besides, tis all optional. Wizards can still be massive lazy dingos and drive admins to drink with Summon Guns and Magic.

[Resources]: # Eventual assets such as icons and sounds which you have and want to suggest related to this feature.
Obviously the gauntlet sprite. I can handle the rest.

[Have fun coding this, jackass]: 
The hardest choices require the strongest wills.
![thanos_smile](https://user-images.githubusercontent.com/33744849/45607364-ff00ee80-ba19-11e8-8e11-015e4325a330.jpg)
But really, I'm definitely biting off more than I can chew in one go. I just want Wizard to have a cool item that works as Declaration of War material.

# Be sure to ask Carbonhell to label this as a feature and not -1 it off the bat because it's big. Either ping them on our discord or PM them on our forum. Thank you for sharing your ideas! BRB re-adding shadowling.
@HippieStation/maintainers

**Comment by WarRage333** [comment]
tldr

+1

**Comment by Manly-Dwarf** [comment]
tl;dr get cool stones and snap your fingers. declare war.

**Comment by bardofrage** [comment]
Sounds like a big project, would be fun though. +1

**Comment by bubberducky2** [comment]
woah man is that homer simpson? +1

**Comment by HippieBartels** [comment]
+1 sounds cool..

**Comment by Eagle-Eyess** [comment]
+1

**Comment by biggiOS** [comment]
+1 i wanna snap my hand and kill half the station

**Comment by ghost** [comment]
did not read a single word

+1

**Comment by Manly-Dwarf** [comment]
9000 IQ.

**Comment by DaydreamPuppeh** [comment]
Fun isn't something one considers when balancing the universe. But this... does put a smile on my face. +1

**Comment by Manly-Dwarf** [comment]
![thanos](https://user-images.githubusercontent.com/33744849/45650693-192fe080-ba9d-11e8-8e79-363cf9037dc1.png)

**Comment by ExpAntique** [comment]
KBB carrying the hippie dev team as usual 

+10000

**Comment by yoyobatty** [comment]
Accepted.

**Comment by lolmemelord123** [comment]
1+

**Comment by Tranquill6** [comment]
Hell yeah
+1

**Comment by Jujumatic** [comment]
The most complicated thing for this imo would be getting all of the sprites lol

**Comment by AnthonyJNgo** [comment]
PR for this is finished and nearly balanced. Already merged.
